crm_phonecall/models/crm_phonecall.py

```diff
@@ -0,0 +1,228 @@
+        "When the call is over, the status is set to Held.\n"
+        "If the callis not applicable anymore, the status can be set "
+        "to Cancelled.",
+    )
+    email_from = fields.Char(string="Email", help="These people will receive email.")
+    date_open = fields.Datetime(string="Opened", readonly=True)
+    name = fields.Char(string="Call Summary", required=True)
+    active = fields.Boolean(required=False, default=True)
+    duration = fields.Float(help="Duration in minutes and seconds.")
+    tag_ids = fields.Many2many(
+        comodel_name="crm.tag",
+        relation="crm_phonecall_tag_rel",
+        string="Tags",
+        column1="phone_id",
+        column2="tag_id",
+    )
+    partner_phone = fields.Char(string="Phone")
+    priority = fields.Selection(
+        selection=[("0", "Low"), ("1", "Normal"), ("2", "High")],
+        default="1",
+    )
+    date_closed = fields.Datetime(string="Closed", readonly=True)
+    date = fields.Datetime(default=lambda self: fields.Datetime.now())
+    opportunity_id = fields.Many2one(comodel_name="crm.lead", string="Lead/Opportunity")
+    direction = fields.Selection(
+        [("in", "In"), ("out", "Out")], default="out", required=True
+    )
+
+    @api.onchange("partner_id")
+    def _onchange_partner_id(self):
+        """Contact number details should be change based on partner."""
+        if self.partner_id:
+            self.partner_phone = self.partner_id.phone
+
+    @api.onchange("opportunity_id")
+    def _onchange_opportunity_id(self):
+        """Based on opportunity, change contact, tags, partner, team."""
+        if self.opportunity_id:
+            self.team_id = self.opportunity_id.team_id.id
+            self.partner_phone = self.opportunity_id.phone
+            self.partner_id = self.opportunity_id.partner_id.id
+            self.tag_ids = self.opportunity_id.tag_ids.ids
+
+    def write(self, values):
+        """Override to add case management: open/close dates."""
+        if values.get("state"):
+            if values.get("state") == "done":
+                values["date_closed"] = fields.Datetime.now()
+                self.compute_duration()
+            elif values.get("state") == "open":
+                values["date_open"] = fields.Datetime.now()
+                values["duration"] = 0.0
+        return super().write(values)
+
+    def compute_duration(self):
+        """Calculate duration based on phonecall date."""
+        phonecall_dates = self.filtered("date")
+        phonecall_no_dates = self - phonecall_dates
+        for phonecall in phonecall_dates:
+            if phonecall.duration <= 0 and phonecall.date:
+                duration = fields.Datetime.now() - phonecall.date
+                values = {"duration": duration.seconds / 60.0}
+                phonecall.write(values)
+            else:
+                phonecall.duration = 0.0
+        phonecall_no_dates.write({"duration": 0.0})
```

Comment:
```suggestion
        phonecall_no_dates.duration = 0.0
```
ORM now has no problem assigning same val to a recordset